FreeRTOS/FreeRTOS-Kernel
# Add configASSERT_SIDE_EFFECT for assertion with volatile variable

Add configASSERT_SIDE_EFFECT for assertion with volatile variable

Description
-----------
Accessing volatile variable in configASSERT may result in different scheduler behavior for non-debug mode. This PR add a `configASSERT_SIDE_EFFECT` for user to configure this kind of assertion. User may define the `configASSERT_SIDE_EFFECT` with the following example to prevent different scheduler behavior for non-debug mode.
```
#define configASSERT_SIDE_EFFECT( x )
```

Test Steps
-----------
Check with coverity scan.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
